templates/_star_chart_heat_map.html

```diff
@@ -0,0 +1,45 @@
+{% load observations %}
+
+{% for construct, sublevels in dot_matrix.items %}
+
+<div style="overflow-x: scroll;width:100%;">
+    <h4 class="text-align-center" style="margin-bottom: 25px;">{{ construct.abbreviation }}</h4>
+        <table class="table table-bordered">
+            <thead>
+                <tr>
+                    {% for level in construct.sublevels %}
+                    <th scope="col" title="{{ level.description }}" data-toggle="tooltip">
+                        {{ level.short_name }}
+                        {% if level.description|length > 0 %}
+                            <span><i class="fa fa-info-circle"></i></span>
+                        {% endif %}
+                    </th>
+                    {% endfor %}
+                </tr>
+            </thead>
+            <tbody>
+                <tr>
+                    {% for sublevel, observations in sublevels.items %}
+                        {% if observations|length < 3 %}
```

Comment:
This not good idea to keep such logic in template. I think you should pass special dict to this case which contains cell's color. Also, logic of colors should come from model.